samples/matter/common/src/bridge/bridge_shell.cpp

```diff
@@ -20,8 +21,17 @@ static int add_bridged_device(const struct shell *shell, size_t argc, char **arg
 		nodeLabel = argv[2];
 	}
 
-	CHIP_ERROR err =
-		GetBridgeManager().AddBridgedDevice(static_cast<BridgedDevice::DeviceType>(deviceType), nodeLabel);
+	auto *newBridgedDevice = BridgeFactory::GetBridgedDeviceFactory().Create(
```

Comment:
I was wondering if we could improve this design even more. So far we've got `BridgeManager` that is independent from supported device types and we've got generic `bridge_util` module. However in the `bridge_shell` we use the `bridge_device_factory` that is specific implementation and may not be used by the user.

What do you think about adding `BridgedDeviceFactory` and `DataProviderFactory` as a class members of a `BridgeManager`. The members could be initialized in the `app_task.cpp` using the `bridged_device_factory` specific implementation. This way we would use the specific implementation only in the `app_task` and the rest of the code would be completely device-type agnostic. In the shell module we could get factories using some getter method of a `BridgeManager`.

Reply:
I would rather think of `BridgeFactory` as an independent module and avoid placing its objects inside any of the current classes. The idea was that the user was responsible for providing factories with desired device types, and the current implementation can be treated as a template which can be easily extended with further device types. 

The shell module uses what has been put in factories by default, but these can be used in the same manner in other places. If we decide that there might be multiple (customized) factories at once, then one can use `DataProviderFactory` and implement them (or maybe then, we should even consider dropping the idea of the factory per se). We may also rename the current ones as `GetDefaultBridgedDeviceFactory` and `GetDefaultDataProviderFactory`. 

We can discuss it f2f :)

Reply:
Ok, as discussed F2F, let it be in the shell for now.